src/sidebar/util/annotation-metadata.js

```diff
@@ -224,7 +269,7 @@ export function location(annotation) {
       const selectors = targets[i].selector || [];
       for (let k = 0; k < selectors.length; k++) {
         if (selectors[k].type === 'TextPositionSelector') {
-          return selectors[k].start;
+          return /** @type {TextPositionSelector} */ (selectors[k]).start;
```

Comment:
I was a bit surprised by this. Without the casting, I saw this error

```
src/sidebar/util/annotation-metadata.js:271:31 - error TS2339: Property 'start' does not exist on type 'TextQuoteSelector | TextPositionSelector'.
  Property 'start' does not exist on type 'TextQuoteSelector'.

271           return selectors[k].start;
```
It does not appear to be the case that typescript looks at the condition and can infer that the type must be TextPositionSelector.

Reply:
It looks like the inference only works when the expression that returns the object you are checking is "simple". If I change the code to replace `selectors[k]` with a variable, then it works:

```diff
diff --git a/src/sidebar/util/annotation-metadata.js b/src/sidebar/util/annotation-metadata.js
index a69ad2044..58845a8c7 100644
--- a/src/sidebar/util/annotation-metadata.js
+++ b/src/sidebar/util/annotation-metadata.js
@@ -268,8 +268,9 @@ export function location(annotation) {
     for (let i = 0; i < targets.length; i++) {
       const selectors = targets[i].selector || [];
       for (let k = 0; k < selectors.length; k++) {
-        if (selectors[k].type === 'TextPositionSelector') {
-          return /** @type {TextPositionSelector} */ (selectors[k]).start;
+        const selector = selectors[k];
+        if (selector.type === 'TextPositionSelector') {
+          return selector.start;
         }
       }
     }
```

I've seen this happen in other contexts too. I would guess it is because figuring out if two different uses of `selectors[k]` will return the same object requires much smarter analysis (for example, making sure that the array wasn't modified in any way in between).

Reply:
If making the above change, you might as well replace the `for` loop with a `for ... of` loop.

Reply:
okay, thanks for sorting that out. Good to know. I feel like we may need to compile a list of "gotchas" for this TS stuff :)